tokio-util/src/udp/frame.rs

```diff
@@ -69,13 +71,12 @@ impl<C: Decoder + Unpin> Stream for UdpFramed<C> {
             let addr = unsafe {
                 // Convert `&mut [MaybeUnit<u8>]` to `&mut [u8]` because we will be
                 // writing to it via `poll_recv_from` and therefore initializing the memory.
-                let buf: &mut [u8] =
-                    &mut *(pin.rd.bytes_mut() as *mut [MaybeUninit<u8>] as *mut [u8]);
-
-                let res = ready!(Pin::new(&mut pin.socket).poll_recv_from(cx, buf));
+                let buf = &mut *(pin.rd.bytes_mut() as *mut _ as *mut [MaybeUninit<u8>]);
+                let mut read = ReadBuf::uninit(buf);
+                let res = ready!(Pin::new(&mut pin.socket).poll_recv_from(cx, &mut read));
```

Comment:
We should either assert that the `ReadBuf` is not swapped, or at least add a comment that we rely on the main Tokio crate doing the right thing here.

Reply:
Let's go with the assertion